erpnext/stock/doctype/batch/batch.py

```diff
@@ -182,6 +189,54 @@ def get_batches_by_oldest(item_code, warehouse):
 	return batches_dates
 
 
+def get_source_batches(batch_no, processed_entries=None):
+	source_batches = []
+
+	if not processed_entries:
+		processed_entries = []
+
+	# fetch unique stock entries where the given batch was manufactured
+	stock_entries = frappe.get_all("Stock Entry",
+		filters=[
+			["Stock Entry Detail", "batch_no", "=", batch_no],
+			["Stock Entry Detail", "s_warehouse", "=", ""],
+			["Stock Entry Detail", "t_warehouse", "!=", ""]
```

Comment:
We should check if the system has proper indexes for this query to run smoothly in the long run when the db has a lot of Batch Entry records. I would imagine batch_no, s_warehouse and t_warehouse to be indexed tho but would be nice to check.

Reply:
Yeah, that's a good point, I'll add an index if it doesn't exist. Although me and @neilLasrado are still figuring out if this is valid anymore, since the workflow this is for has changed significantly.